metals/src/main/scala/scala/meta/internal/metals/Interruptable.scala

```diff
@@ -0,0 +1,56 @@
+package scala.meta.internal.metals
+
+import java.util.concurrent.CompletableFuture
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import scala.concurrent.Promise
+
+import scala.meta.internal.metals.Interruptable.CancelConnectException
+
+class Interruptable[+T] private (
+    futureIn: Future[T],
+    cancelPromise: Promise[Unit],
```

Comment:
Are we able to forward the cancel request to an underlying process if for example bloopInstall is running?